test/format-test.cc

```diff
@@ -2222,16 +2223,21 @@ template <typename Char, typename... T> void check_enabled_formatters() {
 }
 
 TEST(format_test, test_formatters_enabled) {
+  using custom_string =
+      std::basic_string<char, std::char_traits<char>, mock_allocator<char>>;
+  using custom_wstring = std::basic_string<wchar_t, std::char_traits<wchar_t>,
+                                           mock_allocator<wchar_t>>;
+
   check_enabled_formatters<char, bool, char, signed char, unsigned char, short,
                            unsigned short, int, unsigned, long, unsigned long,
                            long long, unsigned long long, float, double,
                            long double, void*, const void*, char*, const char*,
-                           std::string, std::nullptr_t>();
-  check_enabled_formatters<wchar_t, bool, wchar_t, signed char, unsigned char,
-                           short, unsigned short, int, unsigned, long,
-                           unsigned long, long long, unsigned long long, float,
-                           double, long double, void*, const void*, wchar_t*,
-                           const wchar_t*, std::wstring, std::nullptr_t>();
+                           std::string, custom_string, std::nullptr_t>();
```

Comment:
This is my best guess on where this code change should be tested. I figured that given we already test the `std::string` specialization at runtime, testing only the compilation of `custom_string` might be enough.

Reply:
Looks OK but you need to add `rebind` to `mock_allocator` to fix https://github.com/fmtlib/fmt/actions/runs/8779818010/job/24107542630?pr=3943.

Reply:
Yikes, GCC 4.8.

Added the required member types and validated here: https://godbolt.org/z/a1o7YjY5M (every single one of them is required).
I also updated the PR description accordingly (in case you use it as the commit message).

If you don't plan to support GCC 4.8 much longer (which is 11 years old), it might make sense to add a note to remove these types later.

Reply:
Thanks. There are no plans to drop gcc 4.8 at the moment.